### PR TITLE
Merge sweeper code to develop

### PR DESCRIPTION
**Pull request by nickelization**
This adds support for the TTL property.

### CODE REVIEW

File: src/riak_pb_kv_codec.erl [hunk 1]
```diff
@@ -173,21 +175,24 @@ decode_content_meta(indexes, PbIndexes, _Pb) ->
     Indexes = [decode_pair(E) || E <- PbIndexes],
     [{?MD_INDEX, Indexes}];
 decode_content_meta(deleted, DeletedVal, _Pb) ->
-    [{?MD_DELETED, DeletedVal}].
-
+    [{?MD_DELETED, DeletedVal}];
+decode_content_meta(ttl, TTLVal, _Pb) ->
+    [{?MD_DELETED, TTLVal}].
```
Comment (ian-mi) [inline src/riak_pb_kv_codec.erl]: this should be `?MD_TTL`

File: src/riak_pb_kv_codec.erl [hunk 2]
```diff
@@ -173,21 +175,24 @@ decode_content_meta(indexes, PbIndexes, _Pb) ->
...
-          decode_content_meta(charset, PbC#rpbcontent.charset, PbC) ++
-          decode_content_meta(encoding, PbC#rpbcontent.content_encoding, PbC) ++
-          decode_content_meta(vtag, PbC#rpbcontent.vtag, PbC) ++
-          decode_content_meta(links, PbC#rpbcontent.links, PbC) ++
-          decode_content_meta(last_mod, PbC#rpbcontent.last_mod, PbC) ++
-          decode_content_meta(usermeta, PbC#rpbcontent.usermeta, PbC) ++
-          decode_content_meta(indexes, PbC#rpbcontent.indexes, PbC) ++
-          decode_content_meta(deleted, PbC#rpbcontent.deleted, PbC),
+    MD = decode_content_meta(content_type, PbC#rpbcontent.content_type, PbC) ++
+         decode_content_meta(charset,  PbC#rpbcontent.charset, PbC) ++
```
Comment (ian-mi) [inline src/riak_pb_kv_codec.erl]: Confused by the realignment here. Should either be unaligned or matched with the first line.

**Review by ian-mi — CHANGES_REQUESTED** [review]
Looks good, can this be rebased to remove the merge commits?

Might also want to squash e02c30455605638f5bc75150e293dbcef66069dd and 95edf6ede366a860b8a57fbfea4a22f68d9f5250 into e508790ca3e7b3afd7a5d9804c9fdf20a28c9e37

**Comment by nickelization** [comment]
Thanks for the review, @ian-mi!

He and I discussed this over Slack a bit, but I just wanted to post here to keep the process transparent to anyone looking at this PR. We decided against squashing any of Mikael's commits since he's not working with us anymore and I'd rather not rebase somebody else's commits. We're also keeping the merge commits (rather than rebase them away) since they contain conflict resolutions.

File: src/riak_pb_kv_codec.erl (hunk 2 above)
Comment (ian-mi) [inline src/riak_pb_kv_codec.erl]: It looks to me like the second argument of `decode_content_meta` was aligned on each line but the first.

**Comment by davidx** [comment]
@nickelization If using @thumbot to perform the final merge, please note that you can set the ```merge_method: merge```
to ensure a merge commit, vs squash, rebase.
default being 'squash' 

https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button

File: src/riak_pb_kv_codec.erl (hunk 2 above)
Comment (nickelization) [inline src/riak_pb_kv_codec.erl]: Oops, you're right, sorry! I was looking at the indentation changes but didn't notice those other alignment changes. That is definitely screwy, I'll fix that in a minute.

**Comment by ian-mi** [comment]
+1

**Comment by nickelization** [comment]
As discussed with @davidx and @ian-mi, manually hitting the merge button to keep Thumbs from auto-squashing all the history here.

**Pull request merged**